src/auth/jwt/jwt.strategy.ts

```diff
@@ -1,25 +1,38 @@
 import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
 import { PassportStrategy } from '@nestjs/passport';
 import { ExtractJwt, Strategy } from 'passport-jwt';
+import { RedisService } from 'src/common/redis/redis.service';
 import { TokenPayload } from '../interfaces/token-payload.interface';
 
 @Injectable()
 export class AccessTokenStrategy extends PassportStrategy(
   Strategy,
   'accessToken',
 ) {
-  constructor() {
+  constructor(private readonly redisService: RedisService) {
     super({
       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
       ignoreExpiration: false,
       secretOrKey: process.env.JWT_SECRET_KEY,
+      passReqToCallback: true,
     });
   }
 
-  async validate(payload: TokenPayload) {
+  async validate(request: any, payload: TokenPayload) {
     if (payload.sub !== 'accessToken') {
       throw new HttpException('invalid token type', HttpStatus.BAD_REQUEST);
     }
+
+    const tokenFromRequest = request.headers.authorization.split(' ')[1];
+    const tokenInRedis = await this.redisService.getToken(
+      `${payload.userId}-accessToken`,
+    );
+
+    if (!tokenInRedis) {
+      throw new HttpException('token not found', HttpStatus.NOT_FOUND);
+    } else if (tokenInRedis !== tokenFromRequest) {
+      throw new HttpException('token mismatch', HttpStatus.UNAUTHORIZED);
+    }
     return { id: payload.userId };
```

Comment:
```suggestion
    }
    
    return { id: payload.userId };
```
코드간 개행이 있으면 가독성이 좋아집니다.